src/main/java/org/folio/service/orders/flows/update/open/OpenCompositeOrderFlowValidator.java

```diff
@@ -173,43 +178,78 @@ private Future<Void> validateLocationRestrictions(CompositePoLine poLine, List<F
    * The method checking fund location against valid locations and holding locations in POL to identify restricted locations
    * <br> if there is one, will be stored to put in error as parameter.
    * <br> otherwise, order can be opened
-   * @param poLine poLine of order that requested to be open
+   *
+   * @param poLine          poLine of order that requested to be open
    * @param restrictedFunds restricted funds of poLine
-   * @param requestContext requestContext
+   * @param requestContext  requestContext
    * @return Restricted locations
    */
   private Future<Set<String>> extractRestrictedLocationIds(CompositePoLine poLine, Set<Fund> restrictedFunds, RequestContext requestContext) {
-    Set<String> validLocationIds = poLine.getLocations().stream().map(Location::getLocationId).filter(Objects::nonNull).collect(Collectors.toSet());
-    List<String> holdingIds = poLine.getLocations().stream().map(Location::getHoldingId).filter(Objects::nonNull).toList();
-
-    return getPermanentLocationIdsFromHoldings(holdingIds, requestContext)
-      .map(permanentLocationIds -> {
-        logger.info("extractRestrictedLocations:: '{}' restrictedFund(s) is being checked against permanentLocationIds: {} and validLocations: {}",
-          restrictedFunds.size(), permanentLocationIds, validLocationIds);
-        validLocationIds.addAll(permanentLocationIds);
-        Set<String> restrictedLocationsIds = new HashSet<>();
-        for (var fund : restrictedFunds) {
-          List<String> restrictedFundLocationIds = fund.getLocationIds().stream().filter(locationId -> !validLocationIds.contains(locationId)).toList();
-          if (restrictedFundLocationIds.size() == fund.getLocationIds().size()) {
+
+    String currentTenantId = TenantTool.tenantId(requestContext.getHeaders());
+
+    // 1. locations for current tenant
+    var locations = poLine.getLocations()
```

Comment:
if code without 1.2.3.4.5 comments can't explain itself, it will be considered not clean code. I have a feeling that we can reformat code in a way that it explains itself without comment.

1. Lets start with better naming e.g. `locationsForCurrentTenant`

do this the same for e.g `tenantAndHoldingsMap` https://github.com/folio-org/mod-orders/pull/885/files#diff-1817ae44495cc630f4866f6032c4312fe9429d1e1c35ac558dd4ae43641774bdR200

instead of futures. e.g futureLocationsHoldingsList
https://github.com/folio-org/mod-orders/pull/885/files#diff-1817ae44495cc630f4866f6032c4312fe9429d1e1c35ac558dd4ae43641774bdR207

2. This parts can be extracted to method
https://github.com/folio-org/mod-orders/pull/885/files#diff-1817ae44495cc630f4866f6032c4312fe9429d1e1c35ac558dd4ae43641774bdR225-R234

Reply:
Committed too many changes + re-based on master. Please review again.